spec/controllers/tag_wranglers_controller_spec.rb

```diff
@@ -4,8 +4,79 @@
   include LoginMacros
   include RedirectExpectationHelper
 
+  wrangling_roles = %w[superadmin tag_wrangling]
+
   let(:user) { create(:tag_wrangler) }
 
+  shared_examples "denies access to unauthorized admins" do |verb, action, **kwargs|
+    context "when logged in as an admin with no role" do
+      let(:admin) { create(:admin) }
+
+      before do
+        fake_login_admin(admin)
+        send(verb, action, **kwargs)
+      end
+
+      it "redirects with an error" do
+        it_redirects_to_with_error(root_url, "Sorry, only an authorized admin can access the page you were trying to reach.")
+      end
+    end
+
+    (Admin::VALID_ROLES - wrangling_roles).each do |admin_role|
+      context "when logged in as an admin with role #{admin_role}" do
+        let(:admin) { create(:admin, roles: [admin_role]) }
+
+        before do
+          fake_login_admin(admin)
+          send(verb, action, **kwargs)
+        end
+
+        it "redirects with an error" do
+          it_redirects_to_with_error(root_url, "Sorry, only an authorized admin can access the page you were trying to reach.")
+        end
+      end
+    end
+  end
+
+  before(:all) do
+    Role.create!(name: "tag_wrangler")
+  end
+
+  after(:all) do
+    # Clean up when we're done, a few other specs become unhappy otherwise
+    Role.find_by(name: "tag_wrangler").destroy!
+  end
```

Comment:
This can be turned into a `before` (same as `before(:each)`) and be moved it into the `describe "#index"`. That way you can remove the after hook because [no state is leaked](https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/BeforeAfterAll) and it's more clear that this is only required by the index action.